pinot-query-planner/src/main/java/org/apache/pinot/query/type/TypeFactory.java

```diff
@@ -52,13 +52,24 @@ public RelDataType createRelDataTypeFromSchema(Schema schema) {
   }
 
   private RelDataType toRelDataType(FieldSpec fieldSpec) {
+    RelDataType type = createSqlType(getSqlTypeName(fieldSpec));
+    boolean isArray = !fieldSpec.isSingleValueField();
+    if (isArray) {
+      type = createArrayType(type, -1);
+    }
+    boolean isNullable = fieldSpec.getNullable() != null && fieldSpec.getNullable();
```

Comment:
This is the query path (registering for multi-stage engine) I was referring to, which doesn't check the table config. IMO this is the correct way to keep backward compatibility